File: loex_Golang/pkg/client/marketwebsocketclient/tradewebsocketclient.go
package marketwebsocketclient

import (
	"fmt"
	"loex_golang/logging/applogger"
	"loex_golang/pkg/client/websocketclientbase"
)

// Responsible to handle Trade data from WebSocket
type TradeWebSocketClient struct {
	websocketclientbase.WebSocketClientBase
}

// Initializer
func (p *TradeWebSocketClient) Init(host string) *TradeWebSocketClient {
	p.WebSocketClientBase.Init(host)
	return p
}

// Set callback handler
func (p *TradeWebSocketClient) SetHandler(
	connectedHandler websocketclientbase.ConnectedHandler,
	responseHandler websocketclientbase.ResponseHandler) {
	p.WebSocketClientBase.SetHandler(connectedHandler, p.handleMessage, responseHandler)
}

// Request latest 300 trade data
func (p *TradeWebSocketClient) Request(symbol string, clientId string) {
	topic := fmt.Sprintf("market_%s_trade_ticker", symbol)
	req := fmt.Sprintf("{\"action\": \"%s\", \"ch\": \"%s\",\"id\": \"%s\" }", "req", topic, clientId)

	p.Send(req)

	applogger.Info("WebSocket requested, topic=%s, clientId=%s", topic, clientId)
}

// Subscribe latest completed trade in tick by tick mode
func (p *TradeWebSocketClient) Subscribe(symbol string, clientId string) {
	topic := fmt.Sprintf("market_%s_trade_ticker", symbol)
	sub := fmt.Sprintf("{\"action\": \"%s\",  \"ch\": \"%s\",\"id\": \"%s\" }", "sub", topic, clientId)

	p.Send(sub)

	applogger.Info("WebSocket subscribed, topic=%s, clientId=%s", topic, clientId)
}

// Unsubscribe trade
func (p *TradeWebSocketClient) UnSubscribe(symbol string, clientId string) {
	topic := fmt.Sprintf("market_%s_trade_ticker", symbol)
	unsub := fmt.Sprintf("{\"action\": \"%s\",  \"ch\": \"%s\",\"id\": \"%s\" }", "unsub", topic, clientId)

	p.Send(unsub)

	applogger.Info("WebSocket unsubscribed, topic=%s, clientId=%s", topic, clientId)
}

func (p *TradeWebSocketClient) handleMessage(msg string) (interface{}, error) {
	return msg, nil
}

File: loex_CSharp/Loex.SDK.Core/Client/MarketWebSocketClient/TradeWebSocketClient.cs
using Loex.SDK.Core.Client.WebSocketClientBase;
using Loex.SDK.Core.Log;

namespace Loex.SDK.Core.Client
{
    /// <summary>
    /// Responsible to handle Trade data from WebSocket
    /// </summary>
    public class TradeWebSocketClient : WebSocketV2ClientBase
    { 

        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="host">websockethost</param>
    public TradeWebSocketClient(string host = DEFAULT_HOST)
            :base("", "", host)
        {
        }

        /// <summary>
        /// Request latest 300 trade data
        /// </summary>
        /// <param name="symbol">Trading symbol</param>
        /// <param name="clientId">Client id</param>
        public void Req(string symbol, string clientId = "")
        {
            string topic = $"market_{symbol}_trade_ticker";

            _WebSocket.Send($"{{\"action\": \"req\",\"ch\": \"{topic}\" }}");

            _logger.Log(LogLevel.Info, $"WebSocket requested, topic={topic}, clientId={clientId}");
        }


        /// <summary>
        /// Subscribe latest completed trade in tick by tick mode
        /// </summary>
        /// <param name="symbol">Trading symbol</param>
        /// <param name="clientId">Client id</param>
        public void Subscribe(string symbol, string clientId = "")
        {
            string topic = $"market_{symbol}_trade_ticker";

            _WebSocket.Send($"{{\"action\": \"sub\", \"ch\": \"{topic}\" }}");

            _logger.Log(LogLevel.Info, $"WebSocket subscribed, topic={topic}, clientId={clientId}");
        }

        /// <summary>
        /// Unsubscribe trade
        /// </summary>
        /// <param name="symbol">Trading symbol</param>
        /// <param name="clientId">Client id</param>
        public void UnSubscribe(string symbol, string clientId = "")
        {
            string topic = $"market_{symbol}_trade_ticker";

            _WebSocket.Send($"{{\"action\": \"unsub\", \"ch\": \"{topic}\"}}");

            _logger.Log(LogLevel.Info, $"WebSocket unsubscribed, topic={topic}, clientId={clientId}");
        }
    }
}
